app/src/main/java/com/example/habitsmasher/ui/history/AddHabitEventDialog.java

```diff
@@ -27,69 +27,59 @@
 import com.example.habitsmasher.DatabaseEntity;
 import com.example.habitsmasher.DatePickerDialogFragment;
 import com.example.habitsmasher.HabitEvent;
+import com.example.habitsmasher.HabitEventDialog;
 import com.example.habitsmasher.R;
 
 /**
  * The AddHabitEventDialog
  * Deals with UI and information handling of the add habit event popup
  */
-public class AddHabitEventDialog extends DialogFragment {
-    private static final String TAG = "AddHabitEventDialog";
-
-    // UI elements
-    private HabitEventListFragment _habitEventListFragment;
-    private EditText _eventCommentText;
-    private TextView _eventDateText;
-    private Button _cancelNewEvent;
-    private Button _confirmNewEvent;
-    private ImageView _eventPictureView;
-    private Uri _selectedImage;
+public class AddHabitEventDialog extends HabitEventDialog {
 
-    /**
-     * Default constructor
-     */
-    public AddHabitEventDialog() {
-    }
+    private AddHabitEventDialog _addFragment = this;
 
```

Comment:
Like in the case of AddHabitDialog/EditHabitDialog, need this field so it can be passed into HabitEventVaildator.